src/server/main.py

```diff
@@ -1,30 +1,259 @@
+        return make_response(
+            "Passwords doesn't match", 202, {"WWW-Authenticate": "Passwords doesn't match"}
+        )
+
+    try:
+        connection: Connection = getDatabaseConnection()
+        sql: str = "SELECT * FROM users WHERE email=?"
+        cursor: Cursor = connection.cursor()
+        cursor.execute(sql, (email,))
+        user_information: list = cursor.fetchall()
+        connection.close()
+    except:
+        return jsonify({"message": "Database connection error!"}), 500
+
+    if user_information:
+        return make_response(
+            "User already exists. Please Log in.",
+            202,
+            {"WWW-Authenticate": "User already exists. Please Log in."},
+        )
+
+    try:
+        connection = getDatabaseConnection()
+        sql = "INSERT INTO users(uuid, email, password_hash) VALUES (?, ?, ?)"
+        cursor = connection.cursor()
+        cursor.execute(
+            sql,
+            (
+                str(unique_id),
+                email,
+                generate_password_hash(password),
+            ),
+        )
+        connection.commit()
+        connection.close()
+    except:
+        return jsonify({"message": "Database connection error!"}), 500
+
+    return make_response("Successfully registered.", 201)
+
+
+@app.route("/login", methods=["POST"])
+def login() -> Response:
+    """Login endpoint."""
+    auth: Any = request.json
+
+    email: str = auth["email"]
+    password: str = auth["password"]
+
+    try:
+        connection: Connection = getDatabaseConnection()
+        sql: str = "SELECT uuid, email, password_hash FROM users WHERE email=?"
+        cursor: Cursor = connection.cursor()
+        cursor.execute(sql, (email,))
+        user_information: list = cursor.fetchall()
+        connection.close()
+    except:
+        return jsonify({"message": "Database connection error!"}), 500
+
+    user_uuid: str = ""
+    user_email: str = ""
+    user_password: str = ""
+
+    if not user_information:
+        return make_response(
+            "Could not verify", 401, {"WWW-Authenticate": 'Basic realm ="User does not exist!"'}
+        )
+    else:
+        user_uuid = user_information[0][0]  # uuid
+        user_email = user_information[0][1]  # email
+        user_password = user_information[0][2]  # password
+
+    if check_password_hash(user_password, password):
+        token: str = jwt.encode(
+            {
+                "uuid": user_uuid,
+                "email": user_email,
+                "exp": datetime.utcnow() + timedelta(minutes=30),
+            },
+            app.config["SECRET_KEY"],
+        )
+
+        return make_response(jsonify({"token": token}), 201)
+
+    # returns 403 if password is wrong
+    return make_response(
+        "Could not verify", 403, {"WWW-Authenticate": 'Basic realm ="Wrong Password!"'}
+    )
+
+
+@app.route("/me", methods=["GET"])
+@token_required
+def me(unique_id: str, email: str) -> Response:
```

Comment:
Ta funkcja miała także odświeżać token, tj przedłużać jego ważność.

Reply:
If I'm not mistaken, the only way to refresh a token is to create a new one using the same user credentials. I can implement creating a new token and returning it in this endpoint, but wouldn't it be better to create a new endpoint, e.g. /refresh, which will only be used for refreshing tokens?

Reply:
We should consider using refresh tokens - user connecting to api will obtain auth token and refresh token. Refresh token is used to obtain a new pair of tokens.

Reply:
In my opinion, managing two tokens may be difficult. I have implemented an endpoint called /refresh, which requires the old token for access, revokes it, and returns a new token

Reply:
I guess this approach should be good enough for our purposes